app/src/main/java/com/termux/x11/XrActivity.java

```diff
@@ -183,8 +186,28 @@ public void onDrawFrame(GL10 gl10) {
         }
     }
 
+    @Override
+    public void onWindowFocusChanged(boolean hasFocus) {
+        super.onWindowFocusChanged(hasFocus);
+        this.hasFocus = hasFocus;
+        if (hasFocus) {
+            keyboardDisabled = true;
+            new Handler().postDelayed(() -> keyboardDisabled = false, 250);
+        }
+    }
+
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
+        // Ignore event madness when disabling software keyboard
+        if (keyboardDisabled) {
+            return true;
+        }
+
+        // Bluetooth/OTG keyboards work properly with the standard flow
+        if (hasFocus) {
+            return super.dispatchKeyEvent(event);
+        }
```

Comment:
There is `hasWindowFocus` function, no need to store current window focus value in separate variable.

Reply:
I fixed that in https://github.com/termux/termux-x11/pull/671/commits/27b52aaafe26a25a84fd2218248cd7aa0e672024